File: Java/Java source-code/sumOfPowerOfDeviations.java


public class sumOfPowerOfDeviations {
    public static double sumOfPowerOfDeviations(Double[] data, Integer k, Double c) {
        double sum = 0;
        for (int i = 0; i < data.length; i++) {
            sum += Math.pow(data[i] - c, k);
        }
        return sum;
    }
}

File: C++/sumOfPowerOfDeviations.cpp
#include <iostream>
#include <math.h>

double sumOfPowerOfDeviations(double data[], int k, double c, int n) {
    double sum = 0;
    for(int i = 0; i < n; i++) {
        sum += pow(data[i] - c, k);
        }
    return sum;
}